apps/smtp/src/modules/smtp/configuration/smtp-metadata-manager.ts

```diff
@@ -27,7 +33,13 @@ export class SmtpMetadataManager {
   > {
     this.logger.debug("Fetching SMTP config");
 
-    return fromPromise(this.metadataManager.get(this.metadataKey, this.saleorApiUrl), (e) => {
+    const timeoutedPromise = racePromise({
+      promise: this.metadataManager.get(this.metadataKey, this.saleorApiUrl),
+      timeout: this.pullConfigTimeout,
+      error: new SmtpMetadataManager.TimeoutExceededError("Timeout while fetching metadata"),
+    });
+
+    return fromPromise(timeoutedPromise, (e) => {
```

Comment:
case, when Saleor lags with metadata is likely an error serious enough to go to Sentry and/or Datadog monitor. It violates SLAs

so we should definitely log this

Reply:
Hmm it will be visible in the logs right now, and my goal was to quickly deliver it to prod and test out whether metadata is a real issue here, so do you think Sentry is needed here? I would say DD monitor would be enough